dask/array/utils.py

```diff
@@ -416,8 +416,10 @@ def ones_like_safe(a, shape, **kwargs):
     is supported (requires NumPy >= 1.17), otherwise falls back to
     using the old behavior, returning ``np.ones(shape, **kwargs)``.
     """
+    from .creation import ones_like
+
     try:
-        return np.ones_like(a, shape=shape, **kwargs)
+        return _array_like_safe(np.ones_like, ones_like, a, shape=shape, **kwargs)
```

Comment:
I think this is a neccessary improvement on this function, but @pentschev please correct me if this is wrong.

Reply:
No, this isn't necessary. `ones_like` already existed before we added `like=` to creation functions like `np.asarray`, in fact, the `like=` argument name comes from the `*_like` functions. We have added `like=` to all creation functions, including `np.ones`, but using `np.ones_like` is equivalent to `np.ones(..., like=...)`. Therefore, `ones_like_safe` already works without the change proposed here.

Reply:
Ok thank you. I will think more about what I am doing wrong then. I was getting a segfault without this change which made me think that maybe I was hitting a wierd loop or somthing. I